pkg/bundle/client.go

```diff
@@ -140,17 +141,27 @@ func (bc *bundleClient) GetActiveBundleNamespacedName(ctx context.Context) (type
 	return nn, nil
 }
 
-func (bc *bundleClient) GetBundleList(ctx context.Context) (bundles []api.PackageBundle, err error) {
+func (bc *bundleClient) GetBundleList(ctx context.Context, serverVersion string) (bundles []api.PackageBundle, err error) {
 	var allBundles = &api.PackageBundleList{}
 	err = bc.Client.List(ctx, allBundles, &client.ListOptions{Namespace: api.PackageNamespace})
 	if err != nil {
 		return nil, fmt.Errorf("listing package bundles: %s", err)
 	}
+
 	sortedBundles := allBundles.Items
 	sortFn := func(i, j int) bool {
```

Comment:
If it were me, I'd take this sort function and give it a name for easier testing.

If you think that sorting function might have good reusability, you could consider extracting a function generator.

Something like:

```go
func newBundleSorter(bundles []Bundle, version string) func(i, j) bool {
    return func(i, j) bool {
        if strings.HasPrefix(bundles[i].Name, version) {
            if !strings.HasPrefix(bundles[j].Name, version) {
                return true
            }
        } else if strings.HasPrefix(bundles[j].Name, version) {
            return false
        }
    }
}
```

Or man, I might be off in a rabbit hole here, but what about this…

What if we added a sort interface on the slice type? Syntax might be off a little but:

```go
// bundlesWithServerVersion provides a sort.Sort interface used to sort the
// bundles relative to a server version, such that the bundles that match
// the server version will be first.
type bundlesWithServerVersion struct {
    bundles []PackageBundle
    version string
}

// SortBundlesByServerVersion sorts the bundles in place. Bundles matching
// version will be first.
func SortBundlesByServerVersion(bundles []PackageBundle, version string) {
    sort.Sort(bundlesWithServerVersion{bundles, version})
}

// Len implements sort.Sort.
func (b bundlesWithServerVersion) Len() int {
    return len(b)
}

// Swap implements sort.Sort.
func (b bundlesWithServerVersion) Swap(i, j int) {
    b.bundles[i], b.bundles[j] = b.bundles[j], b.bundles[i]
}

// Less implements sort.Sort.
func (b bundlesWithServerVersion) Less(i, j int) bool {
    iHasPrefix := strings.HasPrefix(b.bundles[i], b.version)
    jHasPrefix := strings.HasPrefix(b.bundles[j], b.version)
    if iHasPrefix && !jHasPrefix {
        return true
    } else if jHasPrefix && !iHasPrefix{
        return false
    }
    return b.bundles[j].LessThan(&b.bundles[i])
}
```

Then the logic would be with the bundle, and anyone could make use of it, just by calling SortBundlesByVersion.

…but maybe that's just overkill. :D